File: September Challenge/Majority Element II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        map<int, int> mp;
        for(int e : nums) mp[e]++;
        
        vector<int> ans;
        auto itr = mp.begin();
        
        while(itr != mp.end()){
            if(itr->second > nums.size() / 3) ans.push_back(itr->first);
            itr++;
        }
        
        
        return ans;
    }
};

File: Company_wise_Mock/Microsoft/Majority Element II.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n = len(nums)
        th = n // 3 + 1
        
        cnt = defaultdict(int)
        for n in nums:
            cnt[n] += 1
        
        ans = set()
        for n in nums:
            if cnt[n] >= th:
                ans.add(n)
        
        return ans
